cli/container.go

```diff
@@ -74,17 +150,14 @@ func createServer(verbose bool, image string, port string, args []string, env []
 
 	containerName := fmt.Sprintf("k3d-%s-server", name)
 
-	containerPort := nat.Port(fmt.Sprintf("%s/tcp", port))
+	apiPortSpec := fmt.Sprintf("0.0.0.0:%s:6443/tcp", port)
```

Comment:
In `commands.go` we set `--https-listen-port` dynamically to whatever you pass in via `--port, -p` flag. We don't respect this here, so we would fail to connect to the API if we set `--port` to something other than `6443`.

Reply:
Right. I'd still like to know if we can fix the target port to the standard port. Just tested it -- am able to create two clusters using the same target port (6443).

Reply:
We can totally fix it to the default port in the container, but then we also need to pin it in `--https-listen-port` in commands.go :+1:

Reply:
Yes, that will simply k3s starting as well... 

For now, I will fix it by preserve the current behavior: 
-       apiPortSpec := fmt.Sprintf("0.0.0.0:%s:6443/tcp", port)
+       apiPortSpec := fmt.Sprintf("0.0.0.0:%s:%s/tcp", port, port)

I will follow up with fixing api-port in a separate PR.

Reply:
I don't think that provides any benefit, k3s is running in a container with all the ports free. We'll have to manipulate kubeconfig.yaml file after we copy it from docker container to achieve forwarding requests to server.

Reply:
Good point. Just updated the branch with the fix. Would you please try again?

Reply:
That works for me :+1: 
However I noticed, that the connection between the nodes is not working anymore :thinking:

Reply:
@iwilltry42 would you please provide an example on what's not working?

Reply:
Nevermind, I cleaned my docker system and it works now :thinking: 
Before, the nodes were not finding each other (not even themselves)